### ISSUE TITLE
some embedded images are broken

### ISSUE THREAD

**Issue by k2soule** [isawnyu/isaw.web#370]
First noticed Tuesday this week (2/19/2019), but could have manifested earlier; SOME images that had been embedded with text in the body of an object, are no longer appearing (including old pages and recent pages). We're still not sure what the common thing is between the images that are broken, and why the images that still work are ok...

Examples:
https://isaw.nyu.edu/news/new-book-bagnalls-everyday-writing-in-the-graeco-roman-east
https://isaw.nyu.edu/news/from-the-field-the-first-excavation-season-at-gird-i-rostam-kurdistan
https://isaw.nyu.edu/news/object-history-and-significance-the-cameo-of-jupiter-chartres
https://isaw.nyu.edu/research/kinikhoyuk/project

Clicking on the link to a missing image goes to a page not found...
https://isaw.nyu.edu/news/resolveuid/988b6813c3fb442497ef7fced4ed67e7/@@images/image/textflow

This page appears to be ok, but we're not sure why: https://isaw.nyu.edu/graduate-studies

**Comment by skleinfeldt** [comment]
@cguardia can you fix this ASAP please? You can use up to 6 hours on it, although I certainly hope it isn't that hard!

**Comment by cguardia** [comment]
@k2soule is this how this page should look:

http://isaw.nyu.edu:8080/isaw/news/from-the-field-the-first-excavation-season-at-gird-i-rostam-kurdistan

Not exactly sure what happened, but I just hit edit image and got that one, so it had to be stored correctly.

**Comment by cguardia** [comment]
@k2soule This one fails because it's pointing at a Tiff image. Should be png or jpeg:

https://isaw.nyu.edu/news/object-history-and-significance-the-cameo-of-jupiter-chartres

**Comment by cguardia** [comment]
@k2soule This one is pointing at an old blogs.nyu.edu image. THis link now redirects to isaw.nyu.edu/news, and the image link is lost. Need to edit the page and add a working image:

https://isaw.nyu.edu/news/new-book-bagnalls-everyday-writing-in-the-graeco-roman-east

**Comment by cguardia** [comment]
For this one, I also edited/saved the selected image, and it worked.

https://isaw.nyu.edu/research/kinikhoyuk/project

**Comment by cguardia** [comment]
@k2soule so, it seems this happened for different reasons for each image. I changed the first and last, but not sure what images to use for the other two.

**Comment by k2soule** [comment]
Thanks for chasing down these different symptoms. It's surprising how different the issues appear to be. Some of them may have been unrelated to the sudden swarm of broken images; so I'm looking for more that follow one particular pattern. Here is what I'm still seeing:

A page that was working fine a few weeks ago suddenly shows a broken image icon in place of one or more images, and when I hover over the link, it's showing a long URL ending in 
".../@@images/image/textflow"
When I delete that from the end of the URL (editing in plain text), the image loads; as in this example where I made the change in the working copy:  https://isaw.nyu.edu/support-isaw/friends
 
I'm still finding more pages with this problem all over the site...

**Comment by k2soule** [comment]
aha! I think I've figured out part of it: the term "textflow" might refer to a particular preset image display size, which appears to no longer exist.

**Comment by k2soule** [comment]
Yep, if I go in and edit the html, replacing just the word "textflow" with another image size, like "preview" or "mini", the image loads properly.

So, when did we kill the "textflow" image format option?

**Comment by skleinfeldt** [comment]
If I recall correctly, it's pretty easy to add an image size on the control panel, I don't remember where it is but hunt a bit and you should find it @k2soule. If "textflow" is actually there then we need to figure out what's wrong with it. If it is missing hopefully recreating it will fix things.

**Comment by k2soule** [comment]
Good point; I'm going to try re-creating that image size on staging, and see what happens.

**Comment by k2soule** [comment]
Curiously, adding a new image scaling option for "textflow" in the @@imaging-controlpanel on staging did not appear to fix the problem. However, renaming one of the existing image scaling options did help. It *only* worked with one particular size, though: renaming "thumb" to "textflow" allowed the image to display on the page that I'm using to test. https://isaw.jazkarta.com/support-isaw/friends

That trick didn't work with any of the other sizes that I tried. Thumbnail is not the correct size for the images that had been set to display at "textflow" scale, but it's an interesting data point.

**Comment by skleinfeldt** [comment]
Curiouser and curiouser. @cguardia I think we need more help here...

**Comment by cguardia** [comment]
@k2soule it's possible to recreate all scales from the ZMI, so we need to:

1) Add the textflow scale with the correct dimensions.
2) Go to the ZMI and recreate the scales.

If you do 1) above, I can take care of 2). This step can take a very long time, so it's better if it's done directly from the Zope port, rather than using the normal site URL, because the proxy will time out.

**Comment by paregorios** [comment]
@k2soule are you aware of any additional work that might have been done on this ticket following the last comment above?

**Comment by paregorios** [comment]
I fixed the problem with Bagnall's book news item from 2010 by downloading a copy of the cover page from the UC press website and inserting it as a new lead image. Deleted the derelict embed for the lost original image.
